### PR TITLE
spoznione sorry

### PR DESCRIPTION
**Pull request by Sasserine**

### CODE REVIEW

**Review by maxster256 — COMMENTED** [review]
W twojej pracy bardzo podoba mi się to, że po treningu modelu pokazujesz cyfry, które były błędnie identyfikowane. Sam o tym nie pomyślałem, a jednak daje to większe pole do analizy. Z drugiej strony szkoda, że nie są tu badane żadne hiperparametry w modelach. <br>
Na następny raz rozważyłbym również pisanie nieco bardziej zwięzłego kodu. Przykładowo, w przedostatniej komórce z kodem (tam gdzie są zapisywane wyniki trenowanych modeli, większość linijek kodu powtarza się, a jedyną różnicę stanowi nazewnictwo zmiennych, więc można byłoby to wrzucić do jednej funkcji, która przy każdym wywołaniu przyjmuje inny zestaw parametrów.

**Review by Piotr-Wojdas — COMMENTED** [review]
Fajnie, że stworzyłaś 3 modele o różnej złożoności aby porównać je. Jedyne czego mi brakuje to jakieś potencjalne zmiany w hiperparametrach, no i wydaję mi sie że jest trochę za mało opisów przy dosyć dużej ścianie kodu. Over all wnioski z analizy są git a sam wynik modelu jest bardzo wysoki.

**Comment by woznin** [comment]
Hej, super praca!
Świetnie, że testy są zrobione na różnych architekturach, kod opakowany w funkcje i rozpisane wnioski.
Z rzeczy, które można dodać:

- Testy hiperparametrów - sprawdzić różne learning rate'y, f. aktywacji, optimizery czy nawet ilość epok. Z narzędzi, które mogę polecić jest np. [Optuna](https://optuna.readthedocs.io/en/stable/tutorial) czy [GridSearch](https://www.w3schools.com/python/python_ml_grid_search.asp)
- Można dodać [Confusion Matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html) lub [Classification report](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html), ale z kolei te metryki nie biorą pod uwagę czasu uczenia, który jest dodany więc też na plus. CM pomógłby nam zrozumieć, które cyfry najczęściej są mylone ze sobą i ewentualnie dostroić model. 
- Możemy dodać [learning curves](https://wandb.ai/mostafaibrahim17/ml-articles/reports/A-Deep-Dive-Into-Learning-Curves-in-Machine-Learning--Vmlldzo0NjA1ODY0), żeby śledzić jak dobrze nasz model się uczył.

Oprócz tego praca naprawdę git!
